File: patterns/message-filter/java-camel/MainTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.util.List;

public class MainTest {
    @Test
    public void filtersImportantMessages() {
        List<String> input = List.of("IMPORTANT: start", "skip this", "IMPORTANT: end");
        List<String> output = Main.filterImportant(input);
        assertEquals(List.of("IMPORTANT: start", "IMPORTANT: end"), output);
    }
}

File: patterns/message-filter/kotlin-spring-boot/MainTest.kt
import kotlin.test.Test
import kotlin.test.assertEquals

class MainTest {
    @Test
    fun filtersImportantMessages() {
        val input = listOf("IMPORTANT: start", "skip this", "IMPORTANT: end")
        val output = filterImportant(input)
        assertEquals(listOf("IMPORTANT: start", "IMPORTANT: end"), output)
    }
}
